torch/testing/_internal/common_utils.py

```diff
@@ -940,6 +940,35 @@ def get_comparison_dtype(a, b):
 
     return compare_dtype
 
+def compare_with_reference(torch_fn, ref_fn, sample_input):
+    # Compares torch function with reference function for given sample input (object of SampleInput)
+    # Note: only values are compares, type comparison is not done here
+    try:
+        n_inp, n_args, _ = sample_input.numpy()
+    except TypeError:
+        # TODO: This needs a comment
+        return
+    t_inp, t_args, _ = sample_input.splat()
```

Comment:
Note: post discussion with @kshitij12345 - the kwargs will be used from `sample_inputs` only but modifications will be done for specific ops which have divergence of torch vs ref functions (`torch.add` supports alpha param but `np.add` doesn't).

This PR will be updated with this change soon.

Reply:
I think something like the `alpha` parameter warrants a separate `OpInfo` with a custom reference. For example:

```python
OpInfo(
    ...
    variant_test_name="with_alpha",
    # numpy has no builtin reference for the alpha kwarg, but it is easy enough to emulate
    ref=lambda input, other, *, alpha: np.add(input, np.multiply(alpha, other)),
    ...
)
```

Reply:
Correct: we should write the correct reference using functions from NumPy (and SciPy)

Reply:
Great suggestions, for this PR - I'll probably avoid picking add and add a simple example. This OpInfo entry (for add with alpha) will be good for another PR, I guess.